### PR TITLE
Apply `black` to `astropy.utils`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to `astropy.utils` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/utils/iers/iers.py [hunk 1]
```diff
@@ -785,28 +827,30 @@ def _refresh_table_as_needed(self, mjd):
...
-                    f'failed to download {" and ".join(all_urls)}: {err}.\n'
-                    'A coordinate or time-related '
-                    'calculation might be compromised or fail because the dates are '
-                    'not covered by the available IERS file.  See the '
-                    '"IERS data access" section of the astropy documentation '
-                    'for additional information on working offline.'))
+                warn(
+                    AstropyWarning(
+                        f'failed to download {" and ".join(all_urls)}: {err}.\nA'
+                        ' coordinate or time-related calculation might be compromised'
```
Comment (pllim) [inline astropy/utils/iers/iers.py:847]: Wow, it even moves the whitespace around? I thought I was the nitpicky one.

File: astropy/utils/masked/core.py [hunk 2]
```diff
@@ -1070,40 +1109,46 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
...
-        result = x.sum(axis=axis, dtype=dtype, out=out,
-                       keepdims=keepdims, where=where_final)
+        result = x.sum(
+            axis=axis, dtype=dtype, out=out, keepdims=keepdims, where=where_final
+        )
         n -= ddof
         n = np.maximum(n, 0, out=n)
         result /= n
-        result._mask |= (n == 0)
+        result._mask |= n == 0
```
Comment (pllim) [inline astropy/utils/masked/core.py:1143]: Wut? I actually think parenthesis makes it clearer. Ah, well.

File: astropy/utils/tests/test_data.py [hunk 3]
```diff
@@ -1140,13 +1150,13 @@ def test_read_unicode(filename):
     assert isinstance(contents, bytes)
     x = contents.splitlines()[1]
     assert x == (
-        b"\xff\xd7\x94\xd7\x90\xd7\xa1\xd7\x98\xd7\xa8\xd7\x95\xd7\xa0"
-        b"\xd7\x95\xd7\x9e\xd7\x99 \xd7\xa4\xd7\x99\xd7\x99\xd7\xaa\xd7\x95\xd7\x9f"[1:]
+        b"\xff\xd7\x94\xd7\x90\xd7\xa1\xd7\x98\xd7\xa8\xd7\x95\xd7\xa0\xd7\x95\xd7\x9e\xd7\x99 \xd7\xa4\xd7\x99\xd7\x99\xd7\xaa\xd7\x95\xd7\x9f"[
+            1:
```
Comment (pllim) [inline astropy/utils/tests/test_data.py:1154]: This one turns out a little too ugly but okay...

**Review by pllim — APPROVED** [review]
I only did a cursory glance but William did a detailed look as well.

Did you say the precommit ❌ was expected?

Still not a big fan, but I guess I will eventually get used to this.

I would have preferred this to be done early in the release cycle, not right before RC, but hopefully enough people will do RC testing to smoke out what the CI cannot (we do not have perfect coverage).

I approved but I'll let the people active in this campaign do the merging.

File: astropy/utils/data_info.py [hunk 4]
```diff
@@ -471,8 +498,9 @@ class = Column
                 if hasattr(self, 'info_summary_' + option):
                     option = getattr(self, 'info_summary_' + option)
                 else:
-                    raise ValueError('option={} is not an allowed information type'
-                                     .format(option))
+                    raise ValueError(
+                        f'option={option} is not an allowed information type'
```
Comment (eerovaher) [inline astropy/utils/data_info.py:502]:
```suggestion
                        f'{option=} is not an allowed information type'
```

**Comment by mhvk** [comment]
I hope to get to `utils.iers` and `utils.masked` still...

**Review by mhvk — APPROVED** [review]
OK, happy with masked and iers, with just a few changes to the former.

**Comment by pllim** [comment]
@taldcroft , is this too much? I cannot tell if I should approve or ask for rollback.

Screenshot 2022-11-07 105826

**Comment by WilliamJamieson** [comment]
@pllim I am fine with rolling back the changes if you want.

**Comment by pllim** [comment]
I just don't know where to draw the line. While I welcome clean-ups, I also want the backport to not be horrible for the next year. Which commits to keep/throw, I dunno...

File: astropy/utils/masked/function_helpers.py [hunk 5]
```diff
@@ -84,122 +84,57 @@
 
 MASKED_SAFE_FUNCTIONS |= {
     # built-in from multiarray
-    np.may_share_memory,
-    np.can_cast,
-    np.min_scalar_type,
-    np.result_type,
+    np.may_share_memory, np.can_cast, np.min_scalar_type, np.result_type,
```
[the remark this answers is not in the document]
Comment (WilliamJamieson) [reply astropy/utils/masked/function_helpers.py:87]: I believe the `# fmt: skip` at the end (line 114) is supposed fix this.

**Comment by pllim** [comment]
At this point, I am not sure... as long as CI passes, I'll have to leave it to your good judgement whether all the non-black changes here would hinder backports or not.

**Comment by taldcroft** [comment]
For me the important thing is independent human review of the non-black changes to ensure code correctness. It looks like @mhvk is doing that?  Overall I think the footprint of the non-black changes is small enough that I'm not highly worried about backports.

File: astropy/utils/masked/function_helpers.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mhvk) [reply astropy/utils/masked/function_helpers.py:86]: Yes, I now have emacs set up to do black automatically on save, so I know this line will not be touched again!

**Review by mhvk — APPROVED** [review]
I reviewed the changes to `match` for `pytest` and those are all OK. My own changes only ensured better formatting. So, I think this is ready!

**Comment by pllim** [comment]
OK I moved this to "ready for string normalization" in the project board. Thanks, everyone!

**Comment by WilliamJamieson** [comment]
@pllim and @mhvk please let me know if there are any additional changes I need to make. If this PR is ready please move it to the "Ready to Squash" column in https://github.com/orgs/astropy/projects/10.

**Comment by pllim** [comment]
I moved it to "Ready to squash". Thanks!

**Pull request merged**